### PR TITLE
Implementation of RFC0002

### PR DESCRIPTION
**Pull request by kylef**
#### Tasks
- [x] [Combine API Namepace and Resource Namespace](https://github.com/refractproject/rfcs/blob/master/text/0002-clarity-api-description.md#combine-namespaces)
- [x] [Rename parameters to HREF Variables](https://github.com/refractproject/rfcs/blob/master/text/0002-clarity-api-description.md#href-variables)
- [x] [Clarify Inheritance](https://github.com/refractproject/rfcs/blob/master/text/0002-clarity-api-description.md#clarify-inheritance)
  - [x] HREF Variables and HREF values throughout the document
  - [x] Content types in payloads
  - [x] Methods in transitions
- [x] [Transactions at root](https://github.com/refractproject/rfcs/blob/master/text/0002-clarity-api-description.md#transitions-at-root)
- [x] [Provided Media Types](https://github.com/refractproject/rfcs/blob/master/text/0002-clarity-api-description.md#provided-media-types)
- [x] [Transition Attributes](https://github.com/refractproject/rfcs/blob/master/text/0002-clarity-api-description.md#transition-attributes)

### CODE REVIEW

**Comment by kylef** [comment]
This pull request is ready to review, would appreciate if @smizell and @zdne could give a :+1: or :-1:.

File: namespaces/api-description-namespace.md [hunk 1]
```diff
@@ -315,8 +455,8 @@ HTTP request message.
 
 - `element`: httpRequest (string, fixed)
 - `attributes`
-    - `method` (string) - HTTP request method.
-    - `href` (Href) - A concrete URI for the request.
+    - `method` (string) - HTTP request method. The method value SHOULD be inherited from a parent transition if it is unset.
+    - `href` (Href) - A concrete URI for the request. The href SHOULD be inherited from a expanding the any parent transitions href and  hrefVariables if the value is unset.
```
Comment (smizell) [inline namespaces/api-description-namespace.md]: The added sentence may need refactoring a bit.

**Comment by smizell** [comment]
@kylef looks good to me. Would want Z's thoughts too.

**Comment by zdne** [comment]
@kylef @smizell Looks great, merging. :+1:

File: namespaces/api-description-namespace.md [hunk 2]
```diff
@@ -17,6 +17,8 @@ This document extends [Refract][] [Data Structure Namespace][] to define REST Re
     - [Resource Elements](#resource-elements)
```
Comment (pksunkara) [inline namespaces/api-description-namespace.md:17]: This was renamed below but not here.

File: namespaces/api-description-namespace.md (hunk 1 above)
Comment (pksunkara) [inline namespaces/api-description-namespace.md]: `from a expanding the any parent` is illegible.

**Pull request merged**